packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2817,6 +2828,21 @@ export class ContainerRuntime
 				forcedFullTree,
 			} as const;
 
+			// If validateSummaryBeforeUpload is true, validate that the summary generated by the summarizer nodes is
+			// correct before this summary is uploaded.
+			if (this.validateSummaryBeforeUpload) {
+				const validateResult = this.summarizerNode.validateSummary();
+				if (!validateResult.success) {
+					const { success, ...loggingProps } = validateResult;
+					const error = new RetriableSummaryError(
+						validateResult.reason,
+						validateResult.retryAfterSeconds,
```

Comment:
I wonder if we can make this a promise that waits for the dust to settle that we return upwards - maybe this is a later PR. That way, larger changes that may take longer to summarize won't run into this issue. 1 second probably should be enough.

Reply:
Alternatively, allow retrying up to something like 5 times, and track how many retries / how much time is needed.  Hypothesis would be that 4th or 5th try would never actually be needed, but would be cool to be able to measure that "bottom-out" point.

Reply:
@tyler-cai-microsoft I did not understand what your suggestion helps with. Can you elaborate? Maybe with an example?

@markfields I don't think adding 5 retries will help. We already had this and we got rid of it because of how inefficient it was. If we want to do this, I would go back to the original logic of doing 4 attempts and make things simpler.
Right now, there are too may patches and hacks all over the place which makes reasoning about things so difficult. We should not be adding more.